api/routes/sample.py

```diff
@@ -60,6 +88,46 @@ async def create_new_sample(
         return sample_id_format(internal_id)
 
 
+@router.put(
+    '/{project}/batch', response_model=Dict[str, Any], operation_id='batchUpsertSamples'
+)
+async def batch_upsert_samples(
+    samples: SampleUpsertBatchModel,
+    connection: Connection = get_project_write_connection,
+) -> Dict[str, Any]:
+    """Upserts a list of samples with sequences, and returns the list of internal sample IDs"""
+
+    def upsert_sample(sample):
+        if not sample.id:
+            return create_new_sample(sample, connection)
```

Comment:
It looks like `create_new_sample` creates a transaction within itself. Do the two transactions sort of cancel out, (tbh: I don't hink create_new_sample doesn't need to create it's own transaction),

Reply:
Fixed!